librosa/effects.py

```diff
@@ -568,3 +570,74 @@ def split(y, top_db=60, ref=np.max, frame_length=2048, hop_length=512):
 
     # Stack the results back as an ndarray
     return edges.reshape((-1, 2))
+
+
+def preemphasis(y, coef=0.97, zi=None, return_zf=False):
+    '''Pre-emphasize (high-pass filter) an audio signal:
+
+        y[n] -> y[n] - coef * y[n-1]
+
+
+    Parameters
+    ----------
+    y : np.ndarray
+        Audio signal
+
+    coef : positive number
+        Pre-emphasis coefficient.  Larger values provide sharper filtering.
```

Comment:
Typically in the range (0, 1)
At the limit coef=0, the output signal is unchanged.
At the limit coef=1, the output is the first-order derivative of the input.